pkg/tls/reloader_test.go

```diff
@@ -0,0 +1,246 @@
+
+	certPEM1, keyPEM1 := generateSelfSignedCert(t, "test1.local")
+	err = os.WriteFile(certFile, certPEM1, 0o600)
+	require.NoError(t, err)
+	err = os.WriteFile(keyFile, keyPEM1, 0o600)
+	require.NoError(t, err)
+
+	log := logger.GetLogger("tls-test")
+	reloader, err := NewReloader(certFile, keyFile, log)
+	require.NoError(t, err)
+
+	defer reloader.Stop()
+
+	initialCert, err := reloader.getCertificate(nil)
+	require.NoError(t, err)
+	leafCert, err := x509.ParseCertificate(initialCert.Certificate[0])
+	require.NoError(t, err)
+	assert.Equal(t, "test1.local", leafCert.Subject.CommonName)
+
+	certPEM2, keyPEM2 := generateSelfSignedCert(t, "test2.local")
+	err = os.WriteFile(certFile, certPEM2, 0o600)
+	require.NoError(t, err)
+	err = os.WriteFile(keyFile, keyPEM2, 0o600)
+	require.NoError(t, err)
+
+	err = reloader.reloadCertificate()
+	require.NoError(t, err)
+
+	updatedCert, err := reloader.getCertificate(nil)
+	require.NoError(t, err)
+	leafCert, err = x509.ParseCertificate(updatedCert.Certificate[0])
+	require.NoError(t, err)
+	assert.Equal(t, "test2.local", leafCert.Subject.CommonName)
+
+	tlsConfig := reloader.GetTLSConfig()
+	configCert, err := tlsConfig.GetCertificate(nil)
+	require.NoError(t, err)
+	leafCert, err = x509.ParseCertificate(configCert.Certificate[0])
+	require.NoError(t, err)
+	assert.Equal(t, "test2.local", leafCert.Subject.CommonName)
+}
+
+func TestReloader_FileOperations(t *testing.T) {
+	tempDir := t.TempDir()
+	certFile := filepath.Join(tempDir, "cert.pem")
+	keyFile := filepath.Join(tempDir, "key.pem")
+
+	// Create initial files
+	certPEM, keyPEM := generateSelfSignedCert(t, "initial.local")
+	require.NoError(t, os.WriteFile(certFile, certPEM, 0o600))
+	require.NoError(t, os.WriteFile(keyFile, keyPEM, 0o600))
+
+	// Create reloader
+	log := logger.GetLogger("tls-test")
+	reloader, err := NewReloader(certFile, keyFile, log)
+	require.NoError(t, err)
+	require.NoError(t, reloader.Start())
+	defer reloader.Stop()
+
+	// Test 1: Remove the old files and create new files
+	t.Run("Remove and create new files", func(t *testing.T) {
+		// Remove existing files
+		require.NoError(t, os.Remove(certFile))
+		require.NoError(t, os.Remove(keyFile))
+		time.Sleep(500 * time.Millisecond)
+
+		// Create new files with different content
+		certPEM, keyPEM = generateSelfSignedCert(t, "recreated.local")
+		require.NoError(t, os.WriteFile(certFile, certPEM, 0o600))
+		require.NoError(t, os.WriteFile(keyFile, keyPEM, 0o600))
+		time.Sleep(500 * time.Millisecond)
+
+		// Force reload after file changes
+		require.NoError(t, reloader.reloadCertificate())
+
+		cert, err := reloader.getCertificate(nil)
+		require.NoError(t, err)
+		leafCert, err := x509.ParseCertificate(cert.Certificate[0])
+		require.NoError(t, err)
+		assert.Equal(t, "recreated.local", leafCert.Subject.CommonName)
+	})
+
+	// Test 2: Create files after the reloader starts
+	t.Run("Create files after reloader starts", func(t *testing.T) {
```

Comment:
The loader is created in the parent test. You need to refactor the test case entirely..